report.go

```diff
@@ -99,6 +99,10 @@ func report(files []string, typ, output string, every time.Duration, bucketsStr
 		rep, report = vegeta.NewHDRHistogramPlotReporter(&m), &m
 	default:
 		switch {
+		case strings.HasPrefix(typ, "bench"):
+			name := typ[5:]
+			var m vegeta.Metrics
+			rep, report = vegeta.NewBenchReporter(&m, name), &m
```

Comment:
Rather than allowing users to specify a name for the benchmark report, I think it'd be better to reuse the name of the attack, if there's one. You could take this out of each `vegeta.Result`'s `Attack` field, and extract a unique, sorted list to `vegeta.Metrics`.

Reply:
oh interesting, i wasn't aware of the attack names.
What is the target use case for this field? I would think to identify a whole set of results corresponding to an attack (`vegata plot` seems to use it this way) , but looking at the code, it seems to identify each single http response result. So for let's say rate 50 and duration 10s you'd have 50 times the same attack name? Is there a reason for this?

Do you think that instead of using `&vegeta.Metrics{}` as the vegeta.Report, I should implement a custom implementation of vegeta.Report - perhaps a struct that embeds a `vegeta.Metrics` - such that we can extract the names in the `Add()` method?
Is there ever a reason why we would see distinct attack names? I suppose we could maintain a vegeta.Metrics per name.

Reply:
> What is the target use case for this field? I would think to identify a whole set of results corresponding to an attack (vegata plot seems to use it this way)

That is correct. It's present in each `vegeta.Result` because the output of `vegeta attack` is a streaming format, so each result in the stream needs this information.

> Do you think that instead of using &vegeta.Metrics{} as the vegeta.Report, I should implement a custom implementation of vegeta.Report

No. I think we should add a `Attacks []string` field to `vegeta.Metrics` that contains a unique sorted list of the attack names it has seen. This you could then use in the your report.